### PR TITLE
feat: show risk warning before use wallet

### PR DESCRIPTION
**Pull request by NikoManiac**
closes #

### CODE REVIEW

File: packages/maskbook/src/plugins/Wallet/SNSAdaptor/RiskWarningDialog/RiskWarningBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    const { isConfirmed: isRiskWarningConfirmed, setDialog: setRiskWarningDialog } = useWalletRiskWarningDialog()
+
+    useEffect(() => {
+        if (!isRiskWarningConfirmed || !clicked) return
+
+        const currentBoundaryElement = ref.current as unknown as HTMLDivElement
+        const childNode = currentBoundaryElement.childNodes.length > 0 ? currentBoundaryElement.childNodes[0] : null
+        if (!childNode) return
+
+        const clickableElement = (childNode?.nodeType === 3 ? ref.current : childNode) as HTMLElement
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/SNSAdaptor/RiskWarningDialog/RiskWarningBoundary.tsx:23]:
```suggestion
        const clickableElement = (childNode?.nodeType === Node.TEXT_NODE ? ref.current : childNode) as HTMLElement
```

File: packages/maskbook/src/extension/background-script/CloudService.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { getStorage, setStorage } from './StorageService'
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/background-script/CloudService.ts:1]: You can register service under `PluginServices.Wallet`. Take a look at `packages/maskbook/src/plugins/Wallet/services/index.ts`.

File: packages/maskbook/src/extension/service.ts [hunk 3]
```diff
@@ -32,6 +32,7 @@ export const Services = {
     Ethereum: add(() => import('./background-script/EthereumService'), 'Ethereum'),
     SocialNetwork: add(() => import('./background-script/SocialNetworkService'), 'SocialNetwork'),
     Settings: add(() => import('./background-script/SettingsService'), 'Settings'),
+    Cloud: add(() => import('./background-script/CloudService'), 'Cloud'),
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/service.ts:35]: Not necessary to add a new service entry for this feature. Use PluginServices instead.

File: packages/maskbook/src/plugins/Wallet/SNSAdaptor/RiskWarningDialog/RiskWarningBoundary.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    const handleRiskWarning = (event: React.MouseEvent<HTMLElement>) => {
+        if (!isRiskWarningConfirmed) {
+            event.stopPropagation()
+            setRiskWarningDialog({ open: true, pluginId })
+            setClicked(true)
+        }
+    }
+
+    return (
+        <Box ref={ref} onClickCapture={handleRiskWarning}>
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/SNSAdaptor/RiskWarningDialog/RiskWarningBoundary.tsx:36]: Pay attention to style mutations. A new wrapper node may break original styles. Check each place where you use this component.

**Comment by wenluomask** [comment]
Checked.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**